node/src/genesis.rs

```diff
@@ -1,26 +1,298 @@
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AccountData {
+    pub account: Account,
+    pub seed: Word,
+    pub auth_scheme: String,
+    pub auth_seed: String,
+}
+
+// MAKE GENESIS
+// ===================================================================================================
+
+pub fn make_genesis(
+    output_path: &PathBuf,
+    force: &bool,
+    config_filepath: &PathBuf,
+) -> Result<()> {
+    let output_file_path = Path::new(output_path);
+
+    if !force {
+        match output_file_path.try_exists() {
+            Ok(file_exists) => {
+                if file_exists {
+                    return Err(anyhow!("Failed to generate new genesis file \"{}\" because it already exists. Use the --force flag to overwrite.", output_path.display()));
+                }
+            },
+            Err(err) => {
+                return Err(anyhow!(
+                    "Failed to generate new genesis file \"{}\". Error: {err}",
+                    output_path.display()
+                ));
+            },
+        }
+    }
+
+    let genesis_file_path = Path::new(config_filepath);
+
+    if let Ok(file_exists) = genesis_file_path.try_exists() {
+        if !file_exists {
+            return Err(anyhow!(
+                "The {} file does not exist. It is necessary to initialise the node GenesisState.",
+                genesis_file_path.display()
+            ));
+        }
+    } else {
+        return Err(anyhow!("Failed to open process {} file.", genesis_file_path.display()));
+    }
+
+    let genesis_input: GenesisInput = load_config(genesis_file_path).extract().map_err(|err| {
+        anyhow!("Failed to load {} config file: {err}", genesis_file_path.display())
+    })?;
+    println!("Config file: {} has successfully been loaded.", genesis_file_path.display());
+
+    let accounts = create_accounts(&genesis_input.accounts)?;
+    println!("Accounts have successfully been created.");
+
+    let genesis_state = GenesisState::new(accounts, genesis_input.version, genesis_input.timestamp);
+
+    // Write genesis state as binary format
+    fs::write(output_path, genesis_state.to_bytes()).unwrap_or_else(|_| {
+        panic!("Failed to write genesis state to output file {}", output_path.display())
+    });
+
+    println!("Genesis initialisation successful: {} has been created", output_path.display());
+
+    Ok(())
+}
+
+fn create_account_file(
+    account: &Account,
+    seed: Word,
+    auth_scheme: String,
+    auth_seed: String,
+    index: usize,
+) -> Result<()> {
+    let file_path = PathBuf::from(format!("accounts/account{index}.mac"));
+
+    let account_data = AccountData {
+        account: account.clone(),
+        seed,
+        auth_scheme,
+        auth_seed,
+    };
+
+    let json_data = serde_json::to_string(&account_data)?;
+
+    fs::write(file_path.as_path(), json_data).map_err(|err| {
+        anyhow!("Failed to write account file to {}, Error: {err}", file_path.display())
+    })?;
+
+    Ok(())
+}
```

Comment:
Here, why does `new` return a Result ? 

when you are saying `write()` are you referring to `write_into()` as used in serialisation here: 
https://github.com/0xPolygonMiden/miden-base/blob/1841b99c776722854444669cc2168e2eed49bac2/objects/src/accounts/mod.rs#L173

Or 

Directly as write in "write to a file" as done here with:
```rust
fs::write(file_path.as_path(), json_data).map_err(|err| {
        anyhow!("Failed to write account file to {}, Error: {err}", file_path.display())
    })?;
```

Reply:
> Here, why does `new` return a Result ?

I was thinking to cover the case of when parsing of auth info (scheme and seed) fails.

For the `write()` function, I was thinking that it would be the second option (write to file).